test/WebJobs.Extensions.DurableTask.Analyzers.Test/Entity/DispatchEntityNameAnalyzerTests.cs

```diff
@@ -111,25 +152,58 @@ public class MyEmptyEntity : IMyEmptyEntity
             VerifyCSharpFix(test, ExpectedFix);
         }
 
+        // Tests SyntaxKind.TupleType
         [TestMethod]
-        public void DispatchCall_Tuple()
+        public void DispatchCall_UsingValueTuple()
         {
             var test = @"
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
 using Microsoft.Azure.WebJobs.Extensions.DurableTask;
 
  public class MyEmptyEntity : IMyEmptyEntity
     {
         [FunctionName(""MyEmptyEntity"")]
-        public static Task Run([EntityTrigger] IDurableEntityContext ctx) => ctx.DispatchAsync<Tuple<int, string>>();
+        public static Task Run([EntityTrigger] IDurableEntityContext ctx) => ctx.DispatchAsync<(string, int)>();
     }";
             var expectedDiagnostics = new DiagnosticResult
             {
                 Id = DiagnosticId,
-                Message = string.Format(Resources.DispatchEntityNameAnalyzerMessageFormat, "Tuple<int, string>", "MyEmptyEntity"),
+                Message = string.Format(Resources.DispatchEntityNameAnalyzerMessageFormat, "(string, int)", "MyEmptyEntity"),
+                Severity = Severity,
+                Locations =
+                 new[] {
+                            new DiagnosticResultLocation("Test0.cs", 9, 96)
+                     }
+            };
+
+            VerifyCSharpDiagnostic(test, expectedDiagnostics);
+
+            VerifyCSharpFix(test, ExpectedFix);
+        }
+
+        // Tests interface not defined in user code
+        [TestMethod]
+        public void DispatchCall_ILogger()
+        {
+            var test = @"
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+ public class MyEmptyEntity : IMyEmptyEntity
+    {
+        [FunctionName(""MyEmptyEntity"")]
+        public static Task Run([EntityTrigger] IDurableEntityContext ctx) => ctx.DispatchAsync<ILogger>();
```

Comment:
This scenario might be another one that is tricky to define if customers have interfaces defined in a dependency. I think that is less likely than having activity functions defined in a dependency, so I am fine leaving it unless someone complains.

Reply:
This analyzer is checking that the identifier used in this type argument matches the function name, actually. The tests are just checking that it will flag the type argument (and not throw any exceptions) when the type argument is different SyntaxKinds.